### PR TITLE
Add Dockerfile using esy at runtime

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Problem

The current docker setup is broken for some reason and we want 

## Solution

Since there's a demo coming up we just want some docker setup to work for the testnet, this is a workaround that just uses `esy x ...` at runtime to get things going. Should be good enough until we can fix the static image.

### CODE REVIEW

File: docker/esy.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+RUN esy build
+
+# Copy the static binaries to a known location
+RUN esy cp "#{self.target_dir / 'default' / 'src' / 'bin' / 'deku_cli.exe'}" deku_cli.exe && \
+    esy cp "#{self.target_dir / 'default' / 'src' / 'bin' / 'deku_node.exe'}" deku_node.exe
+RUN strip ./deku_node.exe && strip ./deku_cli.exe
+
+ENV SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
+ENV SSL_CERT_DIR=/etc/ssl/certs
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply docker/esy.Dockerfile:30]: I just wanted to keep it as close as possible but you are correct 👌

File: docker/esy.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Copy the rest of the files
+COPY . .
+
+# TODO: investigate why esy complains that it's not installed if we don't install again
+RUN esy install
+RUN esy build
+
+# Copy the static binaries to a known location
+RUN esy cp "#{self.target_dir / 'default' / 'src' / 'bin' / 'deku_cli.exe'}" deku_cli.exe && \
```
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply docker/esy.Dockerfile:24]: Yes, didn't think about it

**Review by d4hines — APPROVED** [review]
I tested this by building locally and running while linking `/app/data` to the correct files. The node came up successfully (I didn't test a cluster but I think this test was sufficient).

**Review by anchpop — APPROVED** [review]

File: docker/esy.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM esydev/esy:nightly-alpine-latest as builder
+
+RUN apk add libexecinfo-dev && apk --update add ca-certificates
+
+WORKDIR /app
+
+# Add things that doesn't change much, or should bust cache when it does
+COPY ./esy.lock esy.json ./
```
Comment (callistonianembrace) [inline docker/esy.Dockerfile:8]: This doesn't copy the `esy.lock` directory properly (causing esy to re-solve). I used

```
COPY ./esy.lock ./esy.lock
COPY ./esy.json ./esy.json
```

Comment (ulrikstrid) [reply docker/esy.Dockerfile:8]: Can you push this change? This is probably why we need the second `esy install`

Comment (ulrikstrid) [reply docker/esy.Dockerfile:8]: Actually, I can make the change

Comment (ulrikstrid) [reply docker/esy.Dockerfile]: I fixed this, removed the extra `esy install` and changed to `esy build --release` things are much better now :)

**Pull request merged**